src/main/java/com/github/cameltooling/lsp/internal/completion/modeline/CamelKModelineFileType.java

```diff
@@ -0,0 +1,115 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.github.cameltooling.lsp.internal.completion.modeline;
+
+import org.eclipse.lsp4j.CompletionItem;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.regex.Pattern;
+
+/**
+ * Provides configuration utils associated to File Types while handling camel-k modeline
+ *
+ * @author joshiraez
+ */
+public enum CamelKModelineFileType {
```

Comment:
Why using an enum instead of Data structure with classes? Can you precise the advantages?

Reply:
I wanted to use Java enum capabilities instead of using a Data Structure with Data Classes, for type safety and readability mostly. It seemed like a good fit but it's open to be refactored in the future as the class grows (it is very big already).